### DISCUSSION TITLE
txpower cannot be controlled

### DISCUSSION THREAD

**Discussion by aaa147147** [openthread/openthread#11060] [Q&A] [answered]
When I am using an RCP chip, the return value of txpower appears as 0dBm

```
> txpower
otbr-agent: 00:16:23.807 [D] Cli-----------: Input: txpower
otbr-agent: 00:16:23.807 [D] P-SpinelDrive-: Sent spinel frame, flg:0x2, iid:0, tid:11, cmd:PROP_VALUE_GET, key:PHY_TX_POWER
otbr-agent: 00:16:23.807 [D] P-RadioSpinel-: Wait response: tid=11 key=37
otbr-agent: 00:16:23.808 [D] P-SpinelDrive-: Received spinel frame, flg:0x2, iid:0, tid:11, cmd:PROP_VALUE_IS, key:PHY_TX_POWER, power:0
otbr-agent: 00:16:23.808 [D] Cli-----------: Output: 0 dBm
0 dBm
Done
 otbr-agent: 00:16:23.808 [D] Cli-----------: Output: Done
```
So, I set txpower to 100dBm and the result showed no errors
```
> txpower 100
otbr-agent: 00:22:35.043 [D] Cli-----------: Input: txpower 100
otbr-agent: 00:22:35.043 [D] P-SpinelDrive-: Sent spinel frame, flg:0x2, iid:0, tid:10, cmd:PROP_VALUE_SET, key:PHY_TX_POWER, power:100
otbr-agent: 00:22:35.043 [D] P-RadioSpinel-: Wait response: tid=10 key=37
Done
```
When I checked txpower again, the return value showed 100 dBm
```
> txpower
otbr-agent: 00:23:24.620 [D] Cli-----------: Input: txpower
otbr-agent: 00:23:24.620 [D] P-SpinelDrive-: Sent spinel frame, flg:0x2, iid:0, tid:3, cmd:PROP_VALUE_GET, key:PHY_TX_POWER
otbr-agent: 00:23:24.620 [D] P-RadioSpinel-: Wait response: tid=3 key=37
otbr-agent: 00:23:24.625 [D] P-SpinelDrive-: Received spinel frame, flg:0x2, iid:0, tid:3, cmd:PROP_VALUE_IS, key:PHY_TX_POWER, power:100
100 dBm
Done
```
### I think this is problematic, mainly due to the following two issues
**1. The default power should not be 0dBm, which may be an issue with the RCP manufacturer**
**2.Setting txpower to 100 results in successful setting, but this is impossible**

>The power setting acquisition did not take effect, but no error was returned

> When obtaining power again, the result is also 100dBm, which should be the maximum value that the RCP chip can withstand correctly

**Comment by zhanglongxia** [comment, marked as answer]
The power setting method [otPlatRadioSetTransmitPower()](https://github.com/openthread/openthread/blob/main/include/openthread/platform/radio.h#L571) is implemented by the RCP manufacturer. You may need to contact your chip's manufacturer to implement the correct method.

**Reply by aaa147147** [reply]
Thank you for your reply. I will verify this with the RCP manufacturer
